packages/firestore/src/core/transaction_runner.ts

```diff
@@ -113,7 +113,7 @@ export class TransactionRunner<T> {
   }
 
   private isRetryableTransactionError(error: Error): boolean {
```

Comment:
You can make `error` an optional param to communicate the behavior we have observed. This also supports the change you made below for future readers.

`private isRetryableTransactionError(error?: Error): boolean {...`

Reply:
ultranit: Consider using `error: Error | undefined` rather than `error?: Error`.

It's somewhat nonsensical to _not_ specify an argument to a function named "isRetryableTransactionError", and defining the parameter as `error?: Error` allows nonsensical function invocations like this.

In contrast, specifying the argument as `error: Error | undefined` will cause a TypeScript build error if no argument is specified to the function, but will still allow calling it with an explicit `undefined` argument.

IMO the latter is the intended use.

Reply:
e.g. https://www.typescriptlang.org/play/?exactOptionalPropertyTypes=false#code/GYVwdgxgLglg9mABMOcDqMoAsDyAHWBAQwBsAKAUwCcq4qB+ALkQFEa6BKRAbwChEBiCAgDOcEhQB0JOAHNK7KhwDcvAL69eoSISQp0mLADkE+XaQCCJGQHcYYWQFUwAEwrB7FFwtpVmbX0QAH0RwNw8wLy4+QSFRcSkZeWpfFXVNfQxsM3gwUjI0zMMTMBziEitbeydXd09vNK1ULNwCXPywusiXQubi0zbyyrg7B2dw+rJOiKjVXiA